app/views/games/show.html.erb

```diff
@@ -4,7 +4,15 @@
   <% (0..7).each do |row| %>
     <tr>
       <% (0..7).each do |col| %>
-        <td><%= @game.render_piece(row, col) %></td>
+        <td>
+          <% if @game.piece_at_location(row, col).present? %>
+            <%= link_to game_piece_path(game_id: @game, id: @game.piece_at_location(row,col).first) do %>
```

Comment:
This is a good solution.

One idea that could make it even cleaner. :smile:

If you move `render_piece` into a `helper` (`pieces_helper.rb`). You could put the `link_to` inside of the helper. (link_to is available when within a helper)

Since it's a helper, it would also have access to the variables defined in the controller (such as `@game`).

So in the view, you could potentially have something as simple as
```Ruby
<td>
  <%= render_piece(row,col) %>
</td>
```

And the logic for rendering anything (including the link) could be inside of `render_piece`